legistar/events.py

```diff
@@ -109,64 +109,81 @@ def extractRollCall(self, action_detail_url) :
     def ical(self, ical_text):
         value = icalendar.Calendar.from_ical(ical_text)
         return value
-        
+
 
 
 class LegistarAPIEventScraper(LegistarAPIScraper):
 
     def events(self, since_datetime=None):
+        self._events = self._scrapeWebCalendar()
+        self._scraped_events = {}
+
         if since_datetime:
-            params = {'$filter' : "EventLastModifiedUtc gt datetime'{since_datetime}'".format(since_datetime = since_datetime.isoformat())}
+            # Minutes are often published after an event occurs – without a
+            # corresponding event modification. Query all update fields so later
+            # changes are always caught by our scraper, particularly when
+            # scraping narrower windows of time.
+            update_fields = ('EventLastModifiedUtc',
+                             'EventAgendaLastPublishedUTC',
+                             'EventMinutesLastPublishedUTC')
+
+            since_fmt = " gt datetime'{}'".format(since_datetime.isoformat())
+            since_filter = ' or '.join(field + since_fmt for field in update_fields)
+
+            params = {'$filter' : since_filter}
+
         else:
             params = {}
 
         events_url = self.BASE_URL + '/events/'
 
-        web_results = self._scrapeWebCalendar()
-
         for api_event in self.pages(events_url,
                                     params=params,
                                     item_key="EventId"):
-            start = self.toTime(api_event['EventDate'])
+
             # EventTime may be 'None': this try-except block catches those instances.
             try:
                 start_time = time.strptime(api_event['EventTime'], '%I:%M %p')
+
             except TypeError:
                 continue
+
             else:
+                start = self.toTime(api_event['EventDate'])
                 api_event['start'] = start.replace(hour=start_time.tm_hour,
                                                    minute=start_time.tm_min)
-                api_event['status'] = confirmed_or_passed(api_event['start'])
 
-                key = (api_event['EventBodyName'].strip(),
-                       api_event['start'])
+                api_event['status'] = self._event_status(api_event)
 
-                try:
-                    web_event = web_results[key]
-                    yield api_event, web_event
-                except KeyError:
+                # Skip events that do not appear in the web interface.
```

Comment:
We need @reginafcompton to confirm that this logic will work for la metro

Reply:
FWIW I just ran the updated event scrape for LA Metro with a window of 10 days and it finished as expected (or at least, it did not fail).